* Frank Groeneveld <frankgroeneveld@chello.nl>
  Let python-updater use prefix paths

--- python-updater-r1
+++ python-updater-r1
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!@GENTOO_PORTAGE_EPREFIX@/bin/sh
 #
 # A bit of hackery to update everything that is humanly possible
 # that maybe related to an older version of python. This script can
@@ -25,28 +25,28 @@
 
 PKGS_EXCEPTIONS="dev-lang/python sys-apps/portage"
 PKGS_MANUAL="app-office/gnumeric app-office/dia x11-libs/vte"
-LOGFILE="/var/log/python-updater.log"
+LOGFILE="@GENTOO_PORTAGE_EPREFIX@/var/log/python-updater.log"
 
 # portage variables
-PKG_DBDIR=/var/db/pkg
+PKG_DBDIR=@GENTOO_PORTAGE_EPREFIX@/var/db/pkg
 PORTDIR=`portageq portdir`
 PORTDIR_OVERLAYS=`portageq portdir_overlay`
 
 PRETEND=0
 PKGS_TO_REMERGE=""
 PKGS_COUNT_REMERGE=0
-PORTAGE_PYTHON="/usr/bin/python"
+PORTAGE_PYTHON="@GENTOO_PORTAGE_EPREFIX@/usr/bin/python"
 
 # load the gentoo-style info macros, but hack to get around
 # it thinking this is an rc script
 EBUILD="1" 
-source /sbin/functions.sh
+source @GENTOO_PORTAGE_EPREFIX@/sbin/functions.sh
 
 
 
 for old in 2.4 2.3 2.2 2.1; do
 	if [ "${old}" != "${NEW_PY_VER}" ]; then
-		if [ -e /usr/bin/python${old} ] ; then
+		if [ -e "@GENTOO_PORTAGE_EPREFIX@"/usr/bin/python${old} ] ; then
 			OLD_PY_VER=${old}
 			break;
 		fi
@@ -153,7 +153,7 @@
 #
 # Test where portage is, in python2.2 or somewhere else?
 #
-for py in /usr/bin/python /usr/bin/python${OLD_PY_VER} /usr/bin/python${NEW_PY_VER}; do
+for py in "@GENTOO_PORTAGE_EPREFIX@"/usr/bin/python "@GENTOO_PORTAGE_EPREFIX@"/usr/bin/python${OLD_PY_VER} "@GENTOO_PORTAGE_EPREFIX@"/usr/bin/python${NEW_PY_VER}; do
 	if ${py} -c "import portage"; then
 		PORTAGE_PYTHON=${py}
 		break;
@@ -163,11 +163,11 @@
 #
 #
 # Find all packages that have installed something in 
-# /usr/lib/python${OLD_PY_VER}
+# @GENTOO_PORTAGE_EPREFIX@/usr/lib/python${OLD_PY_VER}
 #
 #
-OLD_MODULES_DIRS="/usr/lib/python${OLD_PY_VER} /usr/lib32/python${OLD_PY_VER} /usr/lib64/python${OLD_PY_VER}"
-OLD_INCLUDE_DIR=/usr/include/python${OLD_PY_VER}
+OLD_MODULES_DIRS="@GENTOO_PORTAGE_EPREFIX@/usr/lib/python${OLD_PY_VER} @GENTOO_PORTAGE_EPREFIX@/usr/lib32/python${OLD_PY_VER} @GENTOO_PORTAGE_EPREFIX@/usr/lib64/python${OLD_PY_VER}"
+OLD_INCLUDE_DIR=@GENTOO_PORTAGE_EPREFIX@/usr/include/python${OLD_PY_VER}
 
 eloginfo "Starting Python Updater from ${OLD_PY_VER} to ${NEW_PY_VER} :"
 eloginfo "Searching for packages with files in ${OLD_MODULES_DIRS} .."
